### PR TITLE
Upgrade Python dependencies of Opentrons packages

### PR DESCRIPTION
**Pull request by mcous**
## Overview

Companion PR to Opentrons/opentrons#8228. See monorepo PR for more details and test plan. See Slack builds channel for testing builds via the `python_dep-maintenance` branch.

**Should be merged via rebase-and-merge**

## Review requests

Other than the test plan over in Opentrons/opentrons#8228, the main review request here involves the buildroot commit conventions. So:

- [ ] Each change log entry below corresponds to a commit that does the thing it says it does

## Change log

| package                           | change                                  | notes                       |
| --------------------------------- | --------------------------------------- | --------------------------- |
| package/python3-setuptools        | new package                             | cherry-picked from upstream |
| package/pkg-python                | use host-python3-setuptools when needed | cherry-picked from upstream |
| package/python-setuptools         | add comment about python3-setuptools    | cherry-picked from upstream |
| package/python{3}-setuptools      | bump to version 44.0.0                  | cherry-picked from upstream |
| package/python-setuptools-scm     | bump to version 5.0.1                   | cherry-picked from upstream |
| package/python-toml               | add host variant                        | cherry-picked from upstream |
| package/python-toml               | bump to version 0.10.2                  | cherry-picked from upstream |
| package/python-dotenv             | bump to version 0.19.0                  |                             |
| package/python-typing-extensions  | bump to version 3.10.0.0                |                             |
| package/python-zipp               | new package                             |                             |
| package/python-importlib-metadata | new package                             |                             |
| package/python-click              | bump to version 8.0.1                   |                             |
| package/python-httptools          | bump to version 0.2.0                   |                             |
| package/python-h11                | bump to version 0.12.0                  |                             |
| package/python-idna               | bump to version 3.2                     |                             |
| package/python-six                | bump to version 1.16.0                  |                             |
| package/python-websockets         | bump to version 9.1                     |                             |
| package/python-wsproto            | bump to version 1.0.0                   |                             |
| package/python-asgiref            | new package                             |                             |
| package/python-uvicorn            | bump to version 0.14.0                  |                             |
| package/python-sniffio            | new package                             |                             |
| package/python-anyio              | new package                             |                             |

### CODE REVIEW

**Review by sfoster1 — DISMISSED** [review]
Looks good to me as long as it builds and runs.

File: package/python-sniffio/python-sniffio.mk [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#
+# python-sniffio
+#
+################################################################################
+
+PYTHON_SNIFFIO_VERSION = 1.2.0
+PYTHON_SNIFFIO_SOURCE = sniffio-$(PYTHON_SNIFFIO_VERSION).tar.gz
+PYTHON_SNIFFIO_SITE = https://files.pythonhosted.org/packages/a6/ae/44ed7978bcb1f6337a3e2bef19c941de750d73243fc9389140d62853b686
+PYTHON_SNIFFIO_SETUP_TYPE = setuptools
+PYTHON_SNIFFIO_LICENSE = FIXME: license id couldn't be detected
```
Comment (mcous) [inline package/python-sniffio/python-sniffio.mk]:
```suggestion
PYTHON_SNIFFIO_LICENSE = Apache-2.0
```

**Review by SyntaxColoring — APPROVED** [review]
Functionally seems to all be working, for me!

> the main review request here involves the buildroot commit conventions

This is kind of obnoxious and I don't know how important it is, but It might be a good idea to add notes like this in the messages of the cherry-picked commits:

```
(Opentrons cherry-picked from upstream commit abc123def456)
```

See e.g. [`git cherry-pick -x`](https://git-scm.com/docs/git-cherry-pick#Documentation/git-cherry-pick.txt--x).

**Comment by mcous** [comment]
Latest force push simply amended commit messages to add an `(Opentrons cherry-picked from upstream commit xyz)` message to cherry-picked commits

File: package/python-idna/python-idna.mk [hunk 2]
```diff
@@ -4,11 +4,10 @@
 #
 ################################################################################
 
-PYTHON_IDNA_VERSION = 2.7
+PYTHON_IDNA_VERSION = 3.2
 PYTHON_IDNA_SOURCE = idna-$(PYTHON_IDNA_VERSION).tar.gz
-PYTHON_IDNA_SITE = https://files.pythonhosted.org/packages/65/c4/80f97e9c9628f3cac9b98bfca0402ede54e0563b56482e3e6e45c43c4935
+PYTHON_IDNA_SITE = https://files.pythonhosted.org/packages/cb/38/4c4d00ddfa48abe616d7e572e02a04273603db446975ab46bbcd36552005
 PYTHON_IDNA_LICENSE =  BSD-3-Clause
-PYTHON_IDNA_LICENSE_FILES = LICENSE.rst
```
Comment (sanni-t) [inline package/python-idna/python-idna.mk:11]: Looks like the package has a LICENSE.md file

**Review by sanni-t — COMMENTED** [review]
Changes look good. Thanks for that commit summary table.
Will test once it's done building

**Comment by mcous** [comment]
Latest force push to make sure `idna` commit included update to `README.md` rather than the removal of the previous rst license file

**Review by sanni-t — APPROVED** [review]
Works! All mentioned packages installed without any issues 🚀

**Pull request merged**